internal/wasm/memory.go

```diff
@@ -256,7 +256,7 @@ func (m *MemoryInstance) Grow(delta uint32) (result uint32, ok bool) {
 
 	// If exceeds the max of memory size, we push -1 according to the spec.
 	newPages := currentPages + delta
-	if newPages > m.Max {
+	if newPages > m.Max || int32(delta) < 0 {
```

Comment:
Wouldn't `newPages > m.Max || newPages < currentPages` be clearer, or am I misunderstanding it?

Reply:
yeah clearer